src/environments/cd.js

```diff
@@ -0,0 +1,262 @@
+                        );
+                        break;
+                    case "\\uparrow":
+                    case "\\downarrow": {
+                        const leftLabel = parser.callFunction(
+                            "\\\\cdleft", [firstLabel], []
+                        );
+                        arrow = {
+                            type: "atom",
+                            text: funcName,
+                            mode: "math",
+                            family: "rel",
+                        };
+                        arrow = parser.callFunction("\\Big", [arrow], []);
+                        const rightLabel = parser.callFunction(
+                            "\\\\cdright", [secondLabel], []
+                        );
+                        arrow = {
+                            type: "ordgroup",
+                            mode: "math",
+                            body: [leftLabel, arrow, rightLabel],
+                        };
+                        arrow = parser.callFunction("\\\\cdparent", [arrow], []);
+                        break;
+                    }
+                    case "\\\\cdlongequal":
+                        arrow = parser.callFunction("\\\\cdlongequal", [], []);
+                        break;
+                    case "\\Vert":
+                        arrow = {type: "textord", text: "\\Vert", mode: "math"};
+                        arrow = parser.callFunction("\\Big", [arrow], []);
+                        break;
+                    default:
+                        arrow = {type: "textord", text: " ", mode: "math"};
+                }
+                arrow = {
+                    type: "styling",
+                    body: [arrow],
+                    mode: "math",
+                    style: "display",
+                };
+                row.push(arrow);
+                cell = newCell();
+            }
+        }
+        if (i % 2 === 0) {
+            row.push(cell);
+        } else {
+            row.shift();
+        }
+        row = [];
+        body.push(row);
+    }
+
+    // End row group
+    parser.gullet.endGroup();
+    // End array group defining \\
+    parser.gullet.endGroup();
+
+    // define column separation.
+    const cols = new Array(body[0].length).fill({
+        type: "align",
+        align: "c",
+        pregap: 0.25,  // CD package sets \enskip between columns.
+        postgap: 0.25, // So pre and post each get half an \enskip, i.e. 0.25em.
+    });
+
+    return {
+        type: "array",
+        mode: "math",
+        body,
+        arraystretch: 1,
+        addJot: true,
+        rowGaps: [null],
+        cols,
+        colSeparationType: "CD",
+        hLinesBeforeRow: new Array(body.length + 1).fill([]),
+        isCD: true,
+    };
+}
+
+// The function below are not available for general use.
+// They are here only for internal use by the {CD} environment in placing labels
+// next to vertical arrows.
```

Comment:
The reason we don't have special code for positioning labels above/below horizontal arrows is that the existing arrow commands already support that so we reuse that existing functionality, right?  If that's the case, it might be worth adding a comment about that.

Reply:
Correct, and done.